web/src/bootstrap_typeahead.ts

```diff
@@ -471,6 +494,13 @@ export class Typeahead<ItemType extends string | object> {
         const $active = this.$menu.find(".active").removeClass("active");
         let $next = $active.next();
 
+        // This lets there be a way to not have any item highlighted,
+        // which can be important for e.g. letting the user press enter on
+        // whatever's already in the search box.
+        if (!this.requireHighlight && $active.length && !$next.length) {
+            return;
+        }
```

Comment:
Can you explain the theory behind why returning with no action is the right answer for `prev/next` when no highlight is required? It looks like this implementation has `down` when at the bottom do nothing, not wrap around (the default) or switch to nothing being highlighted.

Reply:
There is an action -- we switch to nothing being highlighted. See `this.$menu.find(".active").removeClass("active");` above.

If you press down twice when at the bottom, the first time it removes the highlight and second time puts it at the top of the list.